server/src/main/java/org/opensearch/index/codec/composite/composite99/Composite99DocValuesWriter.java

```diff
@@ -135,9 +239,9 @@ private void createCompositeIndicesIfPossible(DocValuesProducer valuesProducer,
         // we have all the required fields to build composite fields
         if (compositeFieldSet.isEmpty()) {
             for (CompositeMappedFieldType mappedType : compositeMappedFieldTypes) {
-                if (mappedType.getCompositeIndexType().equals(CompositeMappedFieldType.CompositeFieldType.STAR_TREE)) {
+                if (mappedType instanceof StarTreeMapper.StarTreeFieldType) {
                     try (StarTreesBuilder starTreesBuilder = new StarTreesBuilder(state, mapperService)) {
-                        starTreesBuilder.build(fieldProducerMap);
+                        starTreesBuilder.build(metaOut, dataOut, fieldProducerMap, composite99DocValuesConsumer);
```

Comment:
Major : For every compositeFieldType if you're reinitializing starTreesBuilder how will `fieldNumberAcrossStarTrees` be same across multiple fields ? 

`fieldNumeberAcrossStarTrees` must go from this file into starTreesBuilder

Reply:
Actually looking at this logic again, we are looping through all fields inside star trees builder as well and here also we are looping through all fields.

for all fields
     star trees builder
          for all fields

If we fix this, then we're good. Maybe just call starTreesBuilder once from here, that should be good, no need of any validation.

Let me know if this will not work.

Reply:
As discussed offline, initializing the fieldNumeberAcrossStarTrees in the writer itself, and then passing it as a ctor param in StarTreesBuilder.